src/components/views/Waiter/Waiter.js

```diff
@@ -7,97 +8,125 @@ import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
 import Paper from '@material-ui/core/Paper';
 import Button from '@material-ui/core/Button';
-import { Link } from 'react-router-dom';
 
-const demoContent = [
-  {id: '1', status: 'free', order: null},
-  {id: '2', status: 'thinking', order: null},
-  {id: '3', status: 'ordered', order: 123},
-  {id: '4', status: 'prepared', order: 234},
-  {id: '5', status: 'delivered', order: 345},
-  {id: '6', status: 'paid', order: 456},
-];
+class Waiter extends React.Component {
+  static propTypes = {
+    tables: PropTypes.any,
+    fetchTables: PropTypes.func,
+    fetchTableStatusUpdate: PropTypes.func,
+    loading: PropTypes.shape({
+      active: PropTypes.bool,
+      error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
+    }),
+  }
 
-const renderActions = status => {
-  switch (status) {
-    case 'free':
-      return (
-        <>
-          <Button>thinking</Button>
-          <Button>new order</Button>
-        </>
-      );
-    case 'thinking':
-      return (
-        <Button>new order</Button>
-      );
-    case 'ordered':
-      return (
-        <Button>prepared</Button>
-      );
-    case 'prepared':
+  componentDidMount() {
+    const { fetchTables } = this.props;
+    fetchTables();
+  }
+
+  renderActions(id, status) {
+    const { fetchTableStatusUpdate } = this.props;
+
+    switch (status) {
+      case 'free':
+        return (
+          <>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'thinking')}>thinking</Button>
+            <Button onClick={() => fetchTableStatusUpdate(id, 'ordered')} to={`${process.env.PUBLIC_URL}/waiter/order/`}>new order</Button>
+          </>
+        );
+      case 'thinking':
+        return (
+          <>
```

Comment:
Zasada ogólna jest taka, że jak zwracasz jsx, to musi on posiadać zawsze jeden nadrzędny element. W przypadkach jak jednak chcemy zwrócić listę, to wymyślono ten element pusty <>. W przypadku jak zwracasz jeden element to nie jest ten element pusty potrzebny.